Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -633,59 +656,17 @@ extension RouteController: Router {
     }
     
     public func userIsOnRoute(_ location: CLLocation, status: NavigationStatus?) -> Bool {
-        
-        guard let destination = routeProgress.currentLeg.destination else {
-            preconditionFailure("Route legs used for navigation must have destinations")
-        }
-        
-        // If the user has arrived, do not continue monitor reroutes, step progress, etc
-        if routeProgress.currentLegProgress.userHasArrivedAtWaypoint &&
-            (delegate?.router(self, shouldPreventReroutesWhenArrivingAt: destination) ??
-                DefaultBehavior.shouldPreventReroutesWhenArrivingAtWaypoint) {
-            return true
-        }
-        
-        // If we still wait for the first status from NavNative, there is no need to reroute
-        guard let status = status ?? sharedNavigator.mostRecentNavigationStatus else { return true }
-
-        /// NavNative doesn't support reroutes after arrival.
-        /// The code below is a port of logic from LegacyRouteController
-        /// This should be removed once NavNative adds support for reroutes after arrival. 
-        if status.routeState == .complete {
-            // If the user has arrived and reroutes after arrival should be prevented, do not continue monitor
-            // reroutes, step progress, etc
-            if routeProgress.currentLegProgress.userHasArrivedAtWaypoint &&
-                (delegate?.router(self, shouldPreventReroutesWhenArrivingAt: destination) ??
```

Comment:
Nobody calls this now?

Reply:
RIght. It is now used only by `LegacyRouteController`.

Reply:
This a feature used by our users. This is not a good path going forward, unless we know that we bring this back soon or provide a sensible replacement.